### PR TITLE
Merge latest master into dismiss_t1

### PR DESCRIPTION
**Pull request by effigies**
Hi @madeleinesnyder. This is the result of me running:

```Shell
git remote add madeleinesnyder https://github.com/madeleinesnyder/fmriprep.git
git fetch madeleinesnyder
git checkout -t madeleinesnyder/dismiss_t1
get fetch upstream
git merge upstream/master
```

I then made edits to

* fmriprep/workflows/base.py
* fmriprep/workflows/bold.py
* fmriprep/cli/run.py

Besides merging, I made some more substantial changes that I'll discuss by commenting directly on the "Files changed" tab. (These comments will also show up here in the "Conversation" tab.)

### CODE REVIEW

File: fmriprep/cli/run.py [hunk 1]
```diff
@@ -2,25 +2,20 @@
...
-fMRI run preprocessing workflow
+fMRI preprocessing workflow
 =====
 """
 
-import sys
-import os.path as op
-sys.path.append(op.dirname(__file__))
-import BIDSgenerator
-import glob
```
Comment (effigies) [inline fmriprep/cli/run.py:13]: I have for the most part un-done the `BIDSgenerator`-related changes. These are out-of-scope for this pull request. I think it might be a good idea to try to meet with your lab (I'm off campus, but there are others on-campus who could) to figure out how to get fmriprep to work with your data without changes to our core code, which are going to add friction to any pull requests.

File: fmriprep/cli/run.py [hunk 2]
```diff
@@ -126,35 +118,34 @@ def get_parser():
...
-                        help='use ANTs-based skull-stripping (default, slow))')
-    g_ants.add_argument('--no-skull-strip-ants', dest="skull_strip_ants", action='store_false',
-                        help="don't use ANTs-based skull-stripping (use  AFNI instead, fast)")
-    g_ants.add_argument('--skull-strip-template', dest="skull_strip_template", action='store',
-                        choices = ['oasis', 'scsnl'], default='scsnl',
-                        help="template to use for skull stripping (default: scsnl)")
-    g_ants.set_defaults(skull_strip_ants=False, skull_strip_template='scsnl')
+    g_ants.add_argument('--skull-strip-template', action='store', default='OASIS',
+                        choices=['OASIS', 'NKI'],
+                        help='select ANTs skull-stripping template (default: OASIS))')
```
Comment (effigies) [inline fmriprep/cli/run.py:148]: Here, you had a `scnsl` template, so we've apparently been working in parallel with your lab. This would be a great candidate for another PR, but I've reverted it to our version for this PR.

File: fmriprep/cli/run.py [hunk 3]
```diff
@@ -185,39 +176,20 @@ def get_parser():
...
+        help='Specify UUID of previous run, to include error logs in report. '
+             'No effect without --reports-only.')
     g_other.add_argument('--write-graph', action='store_true', default=False,
                          help='Write workflow graph.')
-    
-    # Smoothing options (MCS 10/6/17)
-    g_smoothing = parser.add_argument_group('Specific options for imaging smoothing after preprocessing')
-    g_smoothing.add_argument('--smoothing-kernel',  action='store',
-                        choices = ['4', '6','8','10'], default=6,
-                        help="smoothing width for already preprocessed imaged (default: 6)")
```
Comment (effigies) [inline fmriprep/cli/run.py:195]: This looks like you are working in parallel with @jdkent (see poldracklab/fmriprep#706). Again, removed to reduce scope of PR.

File: fmriprep/workflows/bold.py [hunk 4]
```diff
@@ -317,112 +346,29 @@ def init_func_preproc_wf(bold_file, ignore, dismiss_t1w, freesurfer,
...
-                        ('outputnode.out_mask', 'inputnode.bold_mask')]),
-                ])
+    workflow.connect([
+        (inputnode, bold_reference_wf, [('bold_file', 'inputnode.bold_file')]),
+        (bold_reference_wf, bold_hmc_wf, [
+            ('outputnode.raw_ref_image', 'inputnode.raw_ref_image')]),
+        (bold_reference_wf, func_reports_wf, [
+            ('outputnode.validation_report', 'inputnode.validation_report')]),
+        (summary, func_reports_wf, [('out_report', 'inputnode.summary_report')]),
+        ])
```
Comment (effigies) [inline fmriprep/workflows/bold.py:357]: In this file, I honestly was a little overwhelmed by trying to figure out all the branching, so I made some fairly major changes. The primary one is that I switched from checking `if dismiss_t1w`/`else` to `if not dismiss_t1w`. The things that could be left unchanged from the current upstream master I left unchanged, and then stuck the necessary changes in `if not dismiss_t1w`.

The other change is that I made `use_syn = False` if `dismiss_t1w` is `True`. This means that a lot of the logic around susceptibility distortion correction could be left alone.

**Comment by effigies** [comment]
Let me know if you have any questions about the changes I've made here.

File: fmriprep/workflows/bold.py (hunk 4 above)
Comment (madeleinesnyder) [reply fmriprep/workflows/bold.py:357]: Ok I'll try this version out. I though bold_reference depended on T1 though, so why is it outside the if not dismiss_t1 loop?

Comment (effigies) [reply fmriprep/workflows/bold.py:357]: `bold_reg_wf` depends on T1. `bold_reference_wf` only depends on the incoming BOLD image.

**Comment by effigies** [comment]
Also, if you want to see what the "Files changed" tab would end up looking like on your PR, once this is merged: https://github.com/poldracklab/fmriprep/compare/master...effigies:dismiss_t1

It might be easier to look at it that way than my changes here, since they include a lot of extraneous things from upstream.

**Comment by madeleinesnyder** [comment]
I'm running into problems with the fixes modules in niworkflows.interfaces.
I've tried downloading the latest nipype and niworkflows but I'm still
stuck. It might take a bit to get it working.

**Comment by effigies** [comment]
To install the latest niworkflows, I'd recommend using (from the fmriprep base directory):

```Python
pip install -r requirements.txt
```

**Comment by oesteban** [comment]
This can be closed @madeleinesnyder (DO NOT MERGE :D)

**Comment by madeleinesnyder** [comment]
Cool. I think everything worked, although I still see my BIDSgenerator.py
file on my local copy of the dismiss_t1 branch

**Comment by oesteban** [comment]
Yes, you will need to execute:

git pull

while in your dismiss_t1 branch.

If you have local changes (git pull won't work), then:

git stash
git pull

Let me know if that worked
